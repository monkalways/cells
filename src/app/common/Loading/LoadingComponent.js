import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
