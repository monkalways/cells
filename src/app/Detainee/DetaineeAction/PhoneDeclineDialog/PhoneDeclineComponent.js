import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';

import Loading from '../../../common/Loading';

const propTypes = {
  cellName: PropTypes.string,
  detainee: PropTypes.shape({
    arrestId: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isDialogOpen: PropTypes.bool.isRequired,
  isUpdatingDetentionLog: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  savePhoneCallDecline: PropTypes.func.isRequired,
  userName: PropTypes.string,
};

const defaultProps = {
  cellName: null,
  userName: null,
};

const PhoneDeclineComponent = ({
  cellName,
  detainee,
  isDialogOpen,
  isUpdatingDetentionLog,
  onClose,
  savePhoneCallDecline,
  userName,
}) => {
  const handleClick = () => {
    const { arrestId } = detainee;
    savePhoneCallDecline(arrestId, cellName, userName);
  };

  return (
    <Dialog
      disableBackdropClick={isUpdatingDetentionLog}
      disableEscapeKeyDown={isUpdatingDetentionLog}
      open={isDialogOpen}
      onClose={onClose}
    >
      <React.Fragment>
        <DialogTitle>
          {`Record that ${detainee.lastName}, ${
            detainee.firstName
          } declined a phone call?`}
        </DialogTitle>
        {isUpdatingDetentionLog && (
          <DialogContent>
            <Loading size={50} />
          </DialogContent>
        )}
        <DialogActions>
          <Button
            id="cancelButton"
            onClick={onClose}
            color="primary"
            disabled={isUpdatingDetentionLog}
          >
            Cancel
          </Button>
          <Button
            id="confirmButton"
            onClick={handleClick}
            color="primary"
            variant="contained"
            autoFocus
            disabled={isUpdatingDetentionLog}
          >
            Confirm
          </Button>
        </DialogActions>
      </React.Fragment>
    </Dialog>
  );
};

PhoneDeclineComponent.propTypes = propTypes;
PhoneDeclineComponent.defaultProps = defaultProps;

export default PhoneDeclineComponent;
