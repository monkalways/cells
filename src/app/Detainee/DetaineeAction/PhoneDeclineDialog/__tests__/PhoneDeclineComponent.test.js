import React from 'react';
import { shallow } from 'enzyme';
import { Dialog } from '@material-ui/core';

import PhoneDeclineComponent from '../PhoneDeclineComponent';
import Loading from '../../../../common/Loading';

describe('PhoneDeclineComponent', () => {
  let cellName;
  let detainee;
  let isDialogOpen;
  let isUpdatingDetentionLog;
  let onClose;
  let savePhoneCallDecline;
  let userName;

  beforeEach(() => {
    cellName = 'C1';
    detainee = {
      arrestId: 'arrestId',
      firstName: 'Bob',
      lastName: 'Saget',
    };
    isDialogOpen = false;
    isUpdatingDetentionLog = false;
    onClose = jest.fn();
    savePhoneCallDecline = jest.fn();
    userName = 'Jay';
  });

  const setup = () => shallow(<PhoneDeclineComponent
    cellName={cellName}
    detainee={detainee}
    isDialogOpen={isDialogOpen}
    isUpdatingDetentionLog={isUpdatingDetentionLog}
    onClose={onClose}
    savePhoneCallDecline={savePhoneCallDecline}
    userName={userName}
  />);

  it('should render content when detention log is not being updated', () => {
    isUpdatingDetentionLog = false;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Loading)).not.toExist();
  });

  it('should render Loading component when detention log is being updated', () => {
    isUpdatingDetentionLog = true;

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Loading)).toExist();
  });

  it('should handle the dialog closing', () => {
    const wrapper = setup();
    wrapper.find(Dialog).simulate('close');
    expect(onClose).toBeCalled();
  });

  it('should handle confirm button click when activity is Cell', () => {
    const { arrestId } = detainee;

    const wrapper = setup();
    wrapper.find('#confirmButton').simulate('click');
    expect(savePhoneCallDecline).toBeCalledWith(arrestId, cellName, userName);
  });

  it('should handle cancel button click', () => {
    const wrapper = setup();
    wrapper.find('#cancelButton').simulate('click');
    expect(onClose).toBeCalled();
  });
});
