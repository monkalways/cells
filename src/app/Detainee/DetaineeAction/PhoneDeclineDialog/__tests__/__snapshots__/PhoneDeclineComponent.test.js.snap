// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDeclineComponent should render Loading component when detention log is being updated 1`] = `
<WithStyles(Dialog)
  disableBackdropClick={true}
  disableEscapeKeyDown={true}
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Record that Saget, Bob declined a phone call?
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(LoadingComponent)
      size={50}
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
      id="cancelButton"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      disabled={true}
      id="confirmButton"
      onClick={[Function]}
      variant="contained"
    >
      Confirm
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`PhoneDeclineComponent should render content when detention log is not being updated 1`] = `
<WithStyles(Dialog)
  disableBackdropClick={false}
  disableEscapeKeyDown={false}
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Record that Saget, Bob declined a phone call?
  </WithStyles(DialogTitle)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      disabled={false}
      id="cancelButton"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      disabled={false}
      id="confirmButton"
      onClick={[Function]}
      variant="contained"
    >
      Confirm
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
